### PR TITLE
Add required tests for `go/netutil`

### PR DESCRIPTION
**Pull request by beingnoble03**
## Description
Adds required tests for `go/netutil`

## Related Issue(s)
- Fixes part of #14931 

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by beingnoble03** [comment]
I don't think it would be helpful to add test for `FullyQualifiedHostname`. Should we add test for it?

File: go/netutil/conn_test.go [hunk 1]
```diff
@@ -77,13 +73,9 @@ func TestReadTimeout(t *testing.T) {
...
 	case err := <-c:
-		if err == nil {
-			t.Fatalf("Expected error, got nil")
-		}
+		assert.Error(t, err)
 
-		if !strings.HasSuffix(err.Error(), "i/o timeout") {
-			t.Errorf("Expected error timeout, got %s", err)
-		}
+		assert.True(t, strings.HasSuffix(err.Error(), "i/o timeout"), "Expected error timeout, got %s", err)
```
Comment (ajm188) [inline go/netutil/conn_test.go:78]: `assert.ErrorContains` here (and other places where we are asserting on the contents of error strings)

File: go/netutil/netutil_test.go [hunk 2]
```diff
@@ -66,9 +64,7 @@ func TestJoinHostPort(t *testing.T) {
 		"[::1]:321":     {host: "::1", port: 321},
 	}
 	for want, input := range table {
-		if got := JoinHostPort(input.host, input.port); got != want {
-			t.Errorf("SplitHostPort(%v, %v) = %#v, want %#v", input.host, input.port, got, want)
-		}
+		assert.Equal(t, want, JoinHostPort(input.host, input.port))
```
Comment (ajm188) [inline go/netutil/netutil_test.go:67]: outside the scope of this PR, but should we delete this function in favor of the stdlib `net.JoinHostPort`? cc @mattlord, the signature is slightly different (we take an `int32` port whereas stdlib takes `string`)

**Review by ajm188 — APPROVED** [review]
LGTM with the one assertion change

File: go/netutil/conn_test.go (hunk 1 above)
Comment (beingnoble03) [reply go/netutil/conn_test.go]: done.

**Review by GuptaManan100 — APPROVED** [review]

**Pull request merged**